### PR TITLE
CORE-19561: Update REST side of key rotation

### PR DESCRIPTION
**Pull request by anien**
Changes:
- Merge POST call to start key rotation for both master wrapping key and vNode wrapping key rotation
- Merge GET call for key rotation status for both types of rotations
- Update paths for POST and GET call to reflect only one of each call is now present
- Use one more generic response for both POST and GET calls. Remove unneeded data class.
- Remove old and new key alias from start key rotation POST call. This is not needed anymore as master wrapping key rotation now checks the default master key alias flag.
- Update tests

Key Rotation REST API now:
Screenshot 2024-02-07 at 13 11 43

### CODE REVIEW

**Review by vkolomeyko — DISMISSED** [review]
LGTM from @corda/rest

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt [hunk 1]
```diff
@@ -23,99 +19,52 @@ import net.corda.rest.response.ResponseEntity
...
-        path = "managed/rotation/{tenantId}",
+        path = "rotation/{tenantId}",
         description = "This method gets the status of the latest key rotation for [tenantId].",
-        responseDescription = "Number of signing keys which need rotating grouped by tenantId's wrapping keys",
+        responseDescription = "Number of keys which need rotating grouped by tenantId or tenantId's wrapping keys",
     )
-    fun getManagedKeyRotationStatus(
-        @RestPathParameter(description = "The tenantId whose wrapping keys are rotating.")
+    fun getKeyRotationStatus(
+        @RestPathParameter(description = "UUID of the virtual node or 'master' for master wrapping key rotation status.")
```
Comment (simon-johnson-r3) [inline components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt]: ...or cluster crypto tenant from the predetermined list

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt [hunk 2]
```diff
@@ -23,99 +19,52 @@ import net.corda.rest.response.ResponseEntity
...
         tenantId: String
-    ): ManagedKeyRotationStatusResponse
+    ): KeyRotationStatusResponse
 
     /**
-     * Initiates the managed key rotation process.
+     * Initiates the master wrapping key or vNode wrapping key rotation process.
```
Comment (simon-johnson-r3) [inline components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt]: or cluster crypto tenant

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt [hunk 3]
```diff
@@ -23,99 +19,52 @@ import net.corda.rest.response.ResponseEntity
...
+     *
+     * @param tenantId UUID of the virtual node or 'master' for master wrapping key rotation.
      *
-     * @param tenantId UUID of the virtual node.
+     * @return Key rotation response where the requestId is the unique ID for the key rotation start request.
```
Comment (simon-johnson-r3) [inline components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt:55]: Am I correct in thinking there is nothing you can do with this, except perhaps compare it with entries in the logs? I wonder whether we should address that point here, to stop people looking for some other api?

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt [hunk 4]
```diff
@@ -23,99 +19,52 @@ import net.corda.rest.response.ResponseEntity
...
      *
      * @throws ServiceUnavailableException If the underlying service for sending messages is not available.
      * @throws InvalidInputDataException If the input parameter is invalid.
      *
      */
     @HttpPOST(
-        path = "managed/rotation/{tenantId}",
-        description = "This method enables to rotate all wrapping keys for tenantId.",
+        path = "rotation/{tenantId}",
+        description = "This method enables to rotate all wrapping keys for tenantId or master wrapping key.",
```
Comment (simon-johnson-r3) [inline components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt]: or cluster crypto tenant (also response)

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt [hunk 5]
```diff
@@ -182,104 +180,97 @@ class KeyRotationRestResourceImpl @Activate constructor(
...
-
         if (!hasPreviousRotationFinished()) {
             throw ForbiddenException("A key rotation operation is already ongoing, a new one cannot be started until it completes.")
         }
 
-        return doManagedKeyRotation(
-            tenantId,
-            publishRequests = { publishToKafka!!.publish(it) }
-        )
+        return if (tenantId == "master") {
```
Comment (simon-johnson-r3) [inline components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt]: can we define "master" somewhere?

File: components/crypto/crypto-rest/src/test/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceTest.kt [hunk 6]
```diff
@@ -303,24 +254,24 @@ class KeyRotationRestResourceTest {
         }
     }
 
-    // We need two wrapping key aliases - oldKeyAlias and newKeyAlias
+    // We need two wrapping key aliases - defaultMasterKeyAlias and anotherMasterKeyAlias
```
Comment (simon-johnson-r3) [inline components/crypto/crypto-rest/src/test/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceTest.kt]: minor but it looks like `anotherMasterKeyAlias` is trying to describe a variable here

File: processors/rest-processor/src/integrationTest/resources/swaggerBaseline-v5_2.json [hunk 7]
```diff
@@ -4351,30 +4351,30 @@
...
       "get" : {
         "tags" : [ "Key Rotation API" ],
         "description" : "This method gets the status of the latest key rotation for [tenantId].",
-        "operationId" : "get_wrappingkey_managed_rotation__tenantid_",
+        "operationId" : "get_wrappingkey_rotation__tenantid_",
         "parameters" : [ {
           "name" : "tenantid",
           "in" : "path",
-          "description" : "The tenantId whose wrapping keys are rotating.",
+          "description" : "UUID of the virtual node or 'master' for master wrapping key rotation status.",
```
Comment (simon-johnson-r3) [inline processors/rest-processor/src/integrationTest/resources/swaggerBaseline-v5_2.json]: or crypto cluster tenant (all of these)

**Review by simon-johnson-r3 — CHANGES_REQUESTED** [review]

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt (hunk 3 above)
Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt:55]: Correct. We use that requestId when submitting kafka message from rest to crypto worker, but we disregard it afterwards. We thought about possibly using it configurable for users at some point. But didn't get that far.
It's easy to remove it and just acknowledge the request and don't return anything. Shall we do that?

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt (hunk 5 above)
Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt]: Will do.

File: components/crypto/crypto-rest/src/test/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceTest.kt (hunk 6 above)
Comment (anien) [reply components/crypto/crypto-rest/src/test/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceTest.kt]: I did renaming twice and forgot to rename the variable in the comment :)

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt (hunk 1 above)
Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt]: Changed.

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt (hunk 2 above)
Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt]: Added.

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt (hunk 3 above)
Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt:55]: I have kept it for now, but we can simply remove it in a separate PR.

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt (hunk 4 above)
Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/KeyRotationRestResource.kt]: Done.

File: processors/rest-processor/src/integrationTest/resources/swaggerBaseline-v5_2.json (hunk 7 above)
Comment (anien) [reply processors/rest-processor/src/integrationTest/resources/swaggerBaseline-v5_2.json]: Fixed.

**Review by simon-johnson-r3 — APPROVED** [review]

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**